enclave_build/src/image_manager.rs

```diff
@@ -0,0 +1,247 @@
+        runtime.block_on(act)
+    }
+
+    /// Extracts the CMD and ENV expressions from the image and returns them each in a
+    /// temporary file
+    fn load(&mut self) -> Result<(NamedTempFile, NamedTempFile)> {
+        let (cmd, env) = self.extract_image()?;
+
+        let cmd_file = crate::docker::write_config(cmd)
+            .map_err(|err| EnclaveBuildError::ExtractError(format!("{:?}", err)))?;
+        let env_file = crate::docker::write_config(env)
+            .map_err(|err| EnclaveBuildError::ExtractError(format!("{:?}", err)))?;
+
+        Ok((cmd_file, env_file))
+    }
+
+    /// Returns architecture information of the image.
+    fn architecture(&mut self) -> Result<String> {
+        let image_name = self.image_name.clone();
+        let act_get_image = async {
+            let image = self.get_image_details(&image_name).await?;
+
+            Ok(format!("{}", image.config.architecture()))
+        };
+
+        let runtime = Runtime::new().map_err(|_| EnclaveBuildError::RuntimeError)?;
+        runtime.block_on(act_get_image)
+    }
+}
+
+impl OciImageManager {
+    /// When calling this constructor, it also tries to create / initialize the cache at
+    /// the default path.\
+    /// If this fails, the ImageManager is still created, but the 'cache'
+    /// field is set to 'None'.
+    pub fn new<S: AsRef<str>>(image_name: S) -> Self {
+        // Add the default ":latest" tag if the image tag is missing
+        let image_name = check_tag(&image_name);
+
+        // The docker daemon is not used, so a local cache needs to be created
+        // Get the default cache root path
+        let root_path = match CacheManager::get_default_cache_root_path() {
+            Ok(path) => path,
+            Err(_) => {
+                // If the cache root path could not be determined, then the cache can not be initialized
+                return Self {
+                    image_name,
+                    cache: None,
+                };
+            }
+        };
+
+        // Try to create / read the cache
+        let cache = match CacheManager::new(&root_path) {
+            Ok(manager) => Some(manager),
+            Err(err) => {
+                // If the cache could not be created, log the error
+                eprintln!("{:?}", err);
+                None
+            }
+        };
+
+        Self { image_name, cache }
+    }
+
+    /// Returns a struct containing image metadata.
+    ///
+    /// If the image is cached correctly, the function tries to fetch the image from the cache.
+    ///
+    /// If the image is not cached or a cache was not created (the 'cache' field is None),
+    /// it pulls the image, caches it (if the 'cache' field is not None) and returns its metadata.
+    ///
+    /// If the pull succeeded but the caching failed, it returns the pulled image metadata.
+    async fn get_image_details<S: AsRef<str>>(&mut self, image_name: S) -> Result<ImageDetails> {
```

Comment:
Modified workflow to avoid unwraps and treat the following case: if we have a missing or invalid cache entry, pull image and store it in the cache.